src/hook/useMediaType.ts

```diff
@@ -9,13 +9,16 @@ export const useMediaType = (query: string): boolean => {
   }, [])
 
   useEffect(() => {
-    const matchQueryList = window.matchMedia(query)
-    setMatches(matchQueryList.matches)
+    // Guard DOM Access for SSR
+    if (typeof window !== 'undefined') {
```

Comment:
I'm wondering if we couldn't introduce a hook, I mean something like `useIsSSR()` for dealing with the SSR state. What do you think?